six/modules/c++/six.sicd/source/ComplexData.cpp

```diff
@@ -141,5 +141,169 @@ bool ComplexData::equalTo(const Data& rhs) const
+            valid = false;
+        }
+        break;
+    case ImageFormationType::PFA:      // 2.12.2
+
+        if (pfa.get())
+        {
+            valid = pfa->validate(*scpcoa, log) && valid;
+            valid = grid->validate(*pfa, *radarCollection, fc, log) && valid;
+        }
+        else
+        {
+            messageBuilder.str("");
+            messageBuilder <<
+                "PFA specified in imageFormation.imageFormationAlgorithm,"
+                << " but member pointer is NULL.";
+            log.error(messageBuilder.str());
+            valid = false;
+        }
+        break;
+    case ImageFormationType::RMA:      // 2.12.3.*
+        if (rma.get())
+        {
+            valid = rma->validate(*collectionInformation, geoData->scp.ecf,
+                    position->arpPoly, fc, log) && valid;
+            valid = grid->validate(*rma, geoData->scp.ecf, position->arpPoly,
+                    fc, log) && valid;
+        }
+        else
+        {
+            messageBuilder.str("");
+            messageBuilder <<
+                "RMA specified in imageFormation.imageFormationAlgorithm,"
+                << " but member pointer is NULL.";
+            log.error(messageBuilder.str());
+            valid = false;
+        }
+        break;
+    default:
+        //2.12.3 (This is not a typo)
+
+        messageBuilder << "Image formation not fully defined." << std::endl
+            << "SICD.ImageFormation.ImageFormAlgo = OTHER or is not set.";
+        log.warn(messageBuilder.str());
+        valid = false;
+        break;
+    }
+
+    return valid;
+}
+
+void ComplexData::fillDerivedFields(bool includeDefault)
+{
+    const scene::PlaneProjectionModel model(
+            scpcoa->slantPlaneNormal(geoData->scp.ecf),
+            grid->row->unitVector,
+            grid->col->unitVector,
+            geoData->scp.ecf,
+            position->arpPoly,
+            grid->timeCOAPoly,
+            scpcoa->look(geoData->scp.ecf));
+
+    geoData->fillDerivedFields(*imageData, model);
+    grid->fillDerivedFields(*collectionInformation, *imageData, *scpcoa);
+    position->fillDerivedFields(*scpcoa);
+    radarCollection->fillDerivedFields();
+    scpcoa->fillDerivedFields(*geoData, *grid, *position);
+
+    double fc = computeFc();
+
+    switch (imageFormation->imageFormationAlgorithm)
+    {
+    case ImageFormationType::RGAZCOMP:
+        if (rgAzComp.get())
+        {
+            rgAzComp->fillDerivedFields(*geoData, *grid, *scpcoa, *timeline);
+            grid->fillDerivedFields(*rgAzComp, *geoData, *scpcoa, fc);
+        }
+        break;
+    case ImageFormationType::PFA:
+        if (pfa.get())
+        {
+            pfa->fillDerivedFields(*position);
+        }
+    case ImageFormationType::RMA:
+        if (rma.get())
+        {
+            rma->fillDerivedFields(*geoData, *position);
+            grid->fillDerivedFields(*rma, geoData->scp.ecf, position->arpPoly);
+        }
```

Comment:
Is gcc or VS giving warnings about not having a `default` case in here?  If so, can you add a
```
default:
    break;
```

Here and anywhere else you're getting a warning?  If you're not getting warnings for this, don't worry about it.